homeassistant/components/openweathermap/config_flow.py

```diff
@@ -62,20 +51,40 @@ async def async_step_user(self, user_input=None):
 
             try:
                 api_online = await _is_owm_api_online(
-                    self.hass, user_input[CONF_API_KEY]
+                    self.hass, user_input[CONF_API_KEY], latitude, longitude
                 )
                 if not api_online:
                     errors["base"] = "invalid_api_key"
             except UnauthorizedError:
                 errors["base"] = "invalid_api_key"
-            except APICallError:
+            except APIRequestError:
                 errors["base"] = "cannot_connect"
 
             if not errors:
                 return self.async_create_entry(
                     title=user_input[CONF_NAME], data=user_input
                 )
-        return self.async_show_form(step_id="user", data_schema=SCHEMA, errors=errors)
+
+        schema = vol.Schema(
+            {
+                vol.Required(CONF_API_KEY): str,
+                vol.Optional(CONF_NAME, default=DEFAULT_NAME): str,
+                vol.Optional(
+                    CONF_LATITUDE, default=self.hass.config.latitude
+                ): cv.latitude,
+                vol.Optional(
+                    CONF_LONGITUDE, default=self.hass.config.longitude
+                ): cv.longitude,
+                vol.Optional(CONF_MODE, default=DEFAULT_FORECAST_MODE): vol.In(
+                    FORECAST_MODES
+                ),
+                vol.Optional(CONF_LANGUAGE, default=DEFAULT_LANGUAGE): vol.In(
+                    LANGUAGES
+                ),
+            }
+        )
+
+        return self.async_show_form(step_id="user", data_schema=schema, errors=errors)
 
     async def async_step_import(self, import_input=None):
         """Set the config entry up from yaml."""
```

Comment:
Do we need to update `CONF_MODE` if importing for older yaml configs?

Reply:
if I remember correctly, set of available modes has been changed

Reply:
Yeah, @freekode is right, this has already been taken care of, see:
https://github.com/home-assistant/core/blob/5e8eb39147e15482d4650f8bedddcdbd3bf87e8b/homeassistant/components/openweathermap/__init__.py#L84-L103

Reply:
That should work for existing entries.  I'm pretty sure the new one will be imported with the new `CONFIG_FLOW_VERSION` so it will never be version 1 and never be migrated.

Reply:
Yeah exactly, so new users will just go through the new flow, whereas existing users will be migrated onto the new flow.

Reply:
We should probably drop yaml support in 0.119 so we don't have to deal with these types of issues.

Reply:
Oh I see what you mean. How do we drop yaml support? And for the time being, where would I need to add support to migrate `freedaily`? Is that inside `async_setup()` -> `_import_configs()`?

Sorry, I'm pretty new to HA development and I'm still not too sure what is triggered when..

Reply:
```
if CONF_MODE not in config:
            config[CONF_MODE] = DEFAULT_FORECAST_MODE
```

If it is defined and not in the list of supported modes defined in `const.py`, we should swap the value to `DEFAULT_FORECAST_MODE` in `async_step_import`

Since config flows require 100% test coverage, we do need an updated test for it as well.